pip/pip-300.md

```diff
@@ -0,0 +1,67 @@
+# Motivation
+
+Dynamic configuration is an important feature in the production environment, the Pulsar supports this feature,
+see `ServiceConfiguration.java`, when the field has `@FieldContext(dynamic = true)` means it is a dynamic configuration,
+and then we can use the `pulsar-admin` to update this field.
+
+The Pulsar has multiple pluggable plugins like authentication provider, authorization provider, and so on. In the
+plugin, we can get the Pulsar configurations and customized configurations, in some scenarios, we need to use
+the `pulsar-admin` to change the values of customized configuration, but the Pulsar doesn't support this operation.
+
+# Goals
+
+## In Scope
+
+The goal of this PIP is to allow the `pulsar-admin` to update the values of customized configuration.
+
+# Detailed Design
+
+## Design & Implementation Details
+
+The `org.apache.pulsar.broker.service.BrokerService.dynamicConfigurationMap` holds the dynamic configurations, so we
+just to add the configurations that need to be dynamically configured to this map, and then we can use
+the `pulsar-admin` to update the values of these configurations.
+
+Add `void registerCustomDynamicConfiguration(String key, Predicate<String> validator)` to the `BrokerService.java`:
+
+```java
+public void registerCustomDynamicConfiguration(String key, Predicate<String> validator) {
```

Comment:
Because this is a kind of public API on the broker side. could you please add non check here? :)

Reply:
Plus, could you please add comments here to declare we will override the old value if the user registers the same key twice?

Reply:
If the user sets the configuration twice, should we consider returning the previous value?

Reply:
Should we consider protecting broker-side dynamic configuration to avoid override by the plugin?